File: frontend/src/types/scene.ts
export interface Scene {
  id: number
  name: string
  description: string
  difficulty: string
  estimated_time: string
  tasks: Task[]
  background: string
  npc_role: string
  learning_objectives: string[]
}

export interface Task {
  id: string
  description: string
  status: 'pending' | 'completed'
  completed_at: string | null
}

export interface TaskProgress {
  completed: number
  total: number
}

export interface Session {
  session_id: string
  scene_info: Scene
  initial_message: string
}

export interface Message {
  message: string
  need_task_check?: boolean
  temperature?: number
}

export interface MessageResponse {
  reply: string
  task_progress?: {
    completed_tasks: string[]
    next_task: string
  }
}

export interface Feedback {
  grammar_corrections: Array<{
    original: string
    correction: string
    explanation: string
  }>
  vocabulary_suggestions: Array<{
    word: string
    context: string
    alternatives: string[]
  }>
  overall_feedback: string
} 

File: backend/app/models/scene.py
from pydantic import BaseModel
from typing import List, Optional

class Scene(BaseModel):
    id: str
    title: str
    description: str
    difficulty: str  # "beginner", "intermediate", "advanced"
    estimated_time: int  # 预计完成时间（分钟）
    tasks: List[str]
    background: str  # 场景背景描述
    npc_role: str  # NPC 角色描述
    learning_objectives: List[str]  # 学习目标 